### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by shahira-sadat**
Hello 👋, this is @shahira-sadat  and @aizjicod 
---
In this PR we did all the requirements according to [project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md)

-  We created data model and validation
-  We created models and relationships
-  We created controllers 
-  We created views for all the pages
-  We created routes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,24 @@
...
+
+* Database initialization
+
+* How to run the test suite
+
+* Services (job queues, cache servers, search engines, etc.)
+
+* Deployment instructions
+
+* ...
```
Comment (Lucas-Masaba) [inline README.md:24]: - [ ] As a requirement for this project, you are required to have a professional `README.md`. Please add al the main headers as seen in the default [template](https://github.com/microverseinc/readme-template). Customize to suit all the features of your project.

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @shahira-sadat and @aizjicod  👋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️ 
- Please read the comment(s) below to see why your project was invalidated.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md (hunk 1 above)
Comment (shahira-sadat) [reply README.md]: Thanks for your feedback. 
Readme file Added ✅

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      <% if @recipe.recipe_foods.length.positive? %>
+        <% @recipe.recipe_foods.each do |recipe_food|%>
+          <tr>
+            <td> <%= recipe_food.food.name%></td>
+            <td> <%= recipe_food.quantity%> <%= recipe_food.food.measurement_unit %></td>
+            <td> $ <%= recipe_food.food.price%></td>
+            <td>
+              <%= button_to 'delete', recipe_food_path(recipe_food.id), method: :delete, class:"btn btn-warning"%>
+            </td>
+          </tr>
```
Comment (elisha2kyakpo1) [inline app/views/recipes/show.html.erb:45]: - [ ] I was expecting to see the foods I add on a recipe show page, but none was shown.

my-food

- [ ] The option to add an ingredient is not well set, a user is given a number field to enter a number, is that number a quantity or amount?

File: app/views/shopping_list/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      <tr>
+        <td> <%= food.food.name %></td>
+        <td> <%= food.quantity %></td>
+        <td> <%= food.food.quantity || 0 %></td>
+        <td> $ <%= food.food.price%></td>
+        <td><%= button_to 'delete', recipe_food_path(food.id), method: :delete, class:"btn btn-warning"%></td>
+      </tr>
+    <%end%>
+  </tbody>
+</table>
```
Comment (elisha2kyakpo1) [inline app/views/shopping_list/index.html.erb:29]: - [ ] When a user clicks on generate list button, the page only shows the title of a recipe with empty data, even if the food was added. This is against the project requirements.

food

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Correct gitflow 👍
- Descriptive PR title and summary text 👍
- The readme is descriptive and looks professional 👍
- The the test are passing 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/shopping_list/index.html.erb (hunk 3 above)
Comment (shahira-sadat) [reply app/views/shopping_list/index.html.erb:29]: @elisha2kyakpo1 could you please check it again?
Its working properly for me 
Screen Shot 2022-11-12 at 12 12 06 AM
Screen Shot 2022-11-12 at 12 11 58 AM

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi Team, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**